core/controllers/subtopic_viewer_test.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for subtopic viewer page"""
+
+from constants import constants
+from core.domain import subtopic_page_domain
+from core.domain import subtopic_page_services
+from core.domain import topic_domain
+from core.domain import user_services
+from core.tests import test_utils
+import feconf
+
+
+class BaseSubtopicViewerControllerTests(test_utils.GenericTestBase):
+
+    user_id = 'user_id'
+
+    def setUp(self):
+        super(BaseSubtopicViewerControllerTests, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.admin = user_services.UserActionsInfo(self.admin_id)
+        self.topic_id = 'topic_id'
+        self.subtopic_id = 'subtopic_id'
```

Comment:
The `subtopic_id` is an integer, so can you give such test values?

Reply:
Done